aiohttp/connector.py

```diff
@@ -1085,14 +1085,18 @@ async def _start_tls_connection(
         # maintainability-wise but this is to be solved separately.
         sslcontext = cast(ssl.SSLContext, self._get_ssl_context(req))
 
+        server_hostname = req.host
+        if req.server_hostname is not None:
+            server_hostname = req.server_hostname
```

Comment:
Should just be able to do:
```suggestion
        server_hostname = req.server_hostname or req.host
```
Maybe worth moving it back inline as well.